app/src/main/java/com/gh4a/Gh4Application.java

```diff
@@ -196,8 +193,7 @@ public void onConfigurationChanged(Configuration newConfig) {
     /* package */ static void trackVisitedUrl(String url) {
         synchronized (Gh4Application.class) {
             if (sHasCrashlytics) {
-                Crashlytics.setString("github-url-" + sNextUrlTrackingPosition, url);
-                Crashlytics.setInt("last-url-position", sNextUrlTrackingPosition);
+                FabricApplication.trackVisitedUrl(url, sNextUrlTrackingPosition);
```

Comment:
As above: Move all the logic into the helper class and only do
```
CrashReportingHelper.trackVisitedUrl(url);
```

here inside the synchronized block.